### PR TITLE
Add autoformatting to CI (for Markdown, YAML, JSON)

### PR DESCRIPTION
**Pull request by fredrik-bakke**
If I didn't mess this up, it will autoformat markdown (and json and yaml) files according to our settings in `.prettierrc.json` on commits to the main branch, i.e. when merging pull requests.

### CODE REVIEW

**Comment by fizruk** [comment]
I've updated the title of the PR, since I originally thought this will autoformat Rzk code somehow.

**Comment by fredrik-bakke** [comment]
Thanks! I wish...

**Comment by fizruk** [comment]
> Thanks! I wish...

I think it should not be too difficult to implement most of the styleguide in a pretty-printer/autoformatter in `rzk` itself, and making it available via the VS Code extension. I think even some checks for the naming conventions can be automated. We'll look into that at some point, I think. But probably not in October :)

File: .github/workflows/rzk.yml
Comment (fizruk) [inline .github/workflows/rzk.yml:29]: I think with the release of rzk-action@v1.1.0 we can drop the `files` input:
```suggestion
```

We could also drop `rzk-version`, but I think it is nice to have it explicitly mention that latest `rzk` is used.

Comment (fredrik-bakke) [reply .github/workflows/rzk.yml]: Does the new default already apply to version 1 of this action?

Comment (fizruk) [reply .github/workflows/rzk.yml]: I assumed it would, but apparently it does not work like that.

Comment (fizruk) [reply .github/workflows/rzk.yml]: I think this is because I had an explicit `v1` tag. I've renamed it to `v1.0.0` just now.

Comment (fizruk) [reply .github/workflows/rzk.yml]: No, apparently, I have to maintain `v1` separately to point to the latest `v1.*.*` tag.

Comment (fizruk) [reply .github/workflows/rzk.yml]: Oh, I see, I need a `v1` branch.

Comment (fizruk) [reply .github/workflows/rzk.yml]: Now `v1` works properly.

Comment (fredrik-bakke) [reply .github/workflows/rzk.yml]: Wait, do you actually want to retroactively change that?

File: .github/workflows/rzk.yml
Comment (fizruk) [inline .github/workflows/rzk.yml:25]: Now this should work:

```suggestion
        uses: rzk-lang/rzk-action@v1
```

**Review by fizruk — APPROVED** [review]

**Comment by fizruk** [comment]
I'm not sure how autoformatting after this is merged will affect current PRs. Perhaps, we should merge #93 first, then this.

File: .github/workflows/rzk.yml
Comment (fizruk) [reply .github/workflows/rzk.yml]: The intent for `v1` was always to point to the latest `v1.*.*`, it's the same with other GitHub Actions, I just did not properly set it up previously.

**Comment by fredrik-bakke** [comment]
There's no rush, but it should minimally affect PRs. The workflow only activates on commits to the `main` branch, i.e. after the PR is merged. It is also configured only to autoformat files touched by the relevant PR.

File: .github/workflows/rzk.yml
Comment (fredrik-bakke) [reply .github/workflows/rzk.yml]: Ah gotcha, that makes sense

**Comment by fizruk** [comment]
> The workflow only activates on commits to the `main` branch, i.e. after the PR is merged. 

Isn't it activated on the merge commit that happens when we merge this PR?

**Comment by fredrik-bakke** [comment]
I assume it does, but this PR doesn't touch any of the files that the other PR touches.

**Comment by fredrik-bakke** [comment]
This is of course assuming the workflow is set up properly.

**Comment by emilyriehl** [comment]
I don't entirely understand what is going on here so I'll leave it for you to decide :)

**Comment by fredrik-bakke** [comment]
> This is of course assuming the workflow is set up properly.


Turns out I didn't 🙃

**Comment by fizruk** [comment]
@fredrik-bakke you can create a separate branch and test it out in it. You should have Write access to create new branches in this repo directly, instead of using a fork.

**Comment by fredrik-bakke** [comment]
Ah, thanks! Yep, this wasn't the best approach

**Comment by fredrik-bakke** [comment]
Turns out the workflow needs a github token with write access, but I don't think I have the privilege to generate those

**Comment by fizruk** [comment]
@fredrik-bakke you should be able to just add something like the following to the job in the workflow file:

```yaml
  permissions: write
```

**Comment by fredrik-bakke** [comment]
Ah, wow, thanks! That seems to have fixed it. I made a new pull request #96.

**Pull request merged**
